test-data/unit/check-flags.test

```diff
@@ -509,3 +509,69 @@ x, y = 1, 2  # type: Unchecked, Unchecked
 [out]
 main:4: error: Type of variable becomes "Any" due to an unfollowed import
 main:6: error: A type on this line becomes "Any" due to an unfollowed import
+
+[case testDisallowAnyExprUnannotatedFunction]
+# flags: --disallow-any=expr
+def g(s):
+    return s  # E: Expressions of type "Any" are disallowed (has type "Any")
+
+g(0)  # E: Parameter 1 to "g" has disallowed type "Any"
+w: int = g(1)  # E: Parameter 1 to "g" has disallowed type "Any"
+
+[case testDisallowAnyExprExplicitAnyParam]
+# flags: --disallow-any=expr
+from typing import Any, List
+def f(s: Any) -> None:
+    pass
+
+def g(s: List[Any]) -> None:
+    pass
+
+f(0)
+
+# because expected type is List[Any] the inferred type of [''] is List[Any], which causes
+# more than one error on the line below
+g([''])
```

Comment:
I've changed my mind on the functions-with-arguments-of-type-Any issue -- I think it's too hard for users to understand, and isn't necessarily in the sprit of this flag.  I think it'd be better to allow those instead.  That should help with a lot of confusing errors.